spec/controllers/callbacks_controller_spec.rb

```diff
@@ -37,5 +37,61 @@
         expect(session).not_to be_empty
       end
     end
+
+    context 'with existing user with invalid token' do
+      let(:user) { create :user_with_invalid_token }
+      before do
+        request.env['omniauth.auth'] = mock_auth(user)
+        request.env['devise.mapping'] = Devise.mappings[:user]
+      end
+
+      it "doesn't create a new User object" do
+        expect do
+          post :github
+        end.to change { User.count }.by(0).and raise_error(MissingMembershipError) { |error|
+          expect(error.message).to eq 'You are not in an organization!'
+        }
+      end
+
+      it "doesn't creates a session" do
+        expect(session).to be_empty
+
+        expect do
+          post :github
+        end.to raise_error(MissingMembershipError) { |error|
+          expect(error.message).to eq 'You are not in an organization!'
+        }
+
+        expect(session).to be_empty
+      end
+    end
+
+    context 'with new user with invalid token' do
```

Comment:
Please use nested contexts (new/existing user vs valid/invalid token).

Basically, something structured like this:

```ruby
context 'with existing user' do
  it 'does something' …
  
  context 'with incorrect token' do
    it 'does something else' …
  end
end

context 'with a new user' do
  it 'creates a user in the DB' …

  context 'with incorrect token' do
    it 'doesn't create a token in the DB' …
  end
end
```

And so on